src/test/java/uk/gov/digital/ho/pttg/application/retry/NameMatchingCandidatesGeneratorTest.java

```diff
@@ -127,4 +128,66 @@ public void shouldHandleMultipleLastNames() {
+        assertThat("The names should be correctly generated in the defined order", names.get(0), is("Arthur OBobbins"));
+        assertThat("The names should be correctly generated in the defined order", names.get(1), is("OBobbins Arthur"));
+        assertThat("The names should be correctly generated in the defined order", names.get(2), is("Arthur Bobbins"));
+        assertThat("The names should be correctly generated in the defined order", names.get(3), is("O Bobbins"));
+        assertThat("The names should be correctly generated in the defined order", names.get(4), is("Bobbins Arthur"));
+        assertThat("The names should be correctly generated in the defined order", names.get(5), is("Bobbins O"));
+        assertThat("The names should be correctly generated in the defined order", names.get(6), is("Arthur O"));
+        assertThat("The names should be correctly generated in the defined order", names.get(7), is("O Arthur"));
+    }
+
+    @Test
+    public void shouldHandleHyphensAndApostrophes() {
+        List<String> names = NameMatchingCandidatesGenerator.generateCandidateNames("Arthur-Brian", "O'Coates");
+
+        assertThat("The number of generated names should be as expected", names.size(), is(14));
+        assertThat("The names should be correctly generated in the defined order", names.get(0), is("ArthurBrian OCoates"));
+        assertThat("The names should be correctly generated in the defined order", names.get(1), is("OCoates ArthurBrian"));
+
+        assertThat("The names should be correctly generated in the defined order", names.get(2), is("Arthur Coates"));
+        assertThat("The names should be correctly generated in the defined order", names.get(3), is("Brian Coates"));
+        assertThat("The names should be correctly generated in the defined order", names.get(4), is("O Coates"));
+        assertThat("The names should be correctly generated in the defined order", names.get(5), is("Arthur Brian"));
+        assertThat("The names should be correctly generated in the defined order", names.get(6), is("Arthur O"));
+        assertThat("The names should be correctly generated in the defined order", names.get(7), is("Brian Arthur"));
+        assertThat("The names should be correctly generated in the defined order", names.get(8), is("O Arthur"));
+        assertThat("The names should be correctly generated in the defined order", names.get(9), is("Coates Arthur"));
+        assertThat("The names should be correctly generated in the defined order", names.get(10), is("Brian O"));
+        assertThat("The names should be correctly generated in the defined order", names.get(11), is("O Brian"));
+        assertThat("The names should be correctly generated in the defined order", names.get(12), is("Coates Brian"));
+        assertThat("The names should be correctly generated in the defined order", names.get(13), is("Coates O"));
+    }
+
+    @Test
+    @Ignore
+    public void thatDiacriticsAreReplaced() {
```

Comment:
I think this test can be removed now diacritics have their own tests

Reply:
It is gone.